src/contracts/core/StakeRootCompendium.sol

```diff
@@ -26,170 +25,139 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
+        uint256 numStrategiesBefore = operatorSetToStrategyAndMultipliers[msg.sender][operatorSet.operatorSetId].length();
+
+        for (uint256 i = 0; i < strategies.length; i++) {
+            require(
+                operatorSetToStrategyAndMultipliers[msg.sender][operatorSet.operatorSetId].remove(address(strategies[i])),
+                "StakeRootCompendium.removeStrategiesAndMultipliers: strategy not found"
+            );
+        }
+
+        _updateTotalStrategies(numStrategiesBefore, operatorSet);
+    }
+
 
-        // remove the strategies and multipliers for the operator set
-        _removeStrategiesAndMultipliers(operatorSet, strategies);
+    function removeOperatorSetFromStakeTree(uint32 operatorSetId) external {
+        _removeOperatorSetFromStakeTree(IAVSDirectory.OperatorSet({avs: msg.sender, operatorSetId: operatorSetId}));
     }
 
     /// @inheritdoc IStakeRootCompendium
     function setOperatorSetExtraData(uint32 operatorSetId, bytes32 extraData) external {
-        (bool exists,, uint224 index) = operatorSetToIndex[msg.sender][operatorSetId].latestCheckpoint();
-        require(exists && index != REMOVED_INDEX, "StakeRootCompendium.setOperatorSetExtraData: operatorSet is not in stakeTree");
         operatorSetExtraDatas[msg.sender][operatorSetId] = extraData;
     }
 
     /// @inheritdoc IStakeRootCompendium
     function setOperatorExtraData(uint32 operatorSetId, address operator, bytes32 extraData) external {
-        (bool exists,, uint224 index) = operatorSetToIndex[msg.sender][operatorSetId].latestCheckpoint();
-        require(exists && index != REMOVED_INDEX, "StakeRootCompendium.setOperatorExtraData: operatorSet is not in stakeTree");
         operatorExtraDatas[msg.sender][operatorSetId][operator] = extraData;
     }
 
     /// POSTING ROOTS AND BLACKLISTING
 
     /// @inheritdoc IStakeRootCompendium
     function verifyStakeRoot(
-        uint32 calculationTimestamp,
+        uint64 calculationBlockNumber,
         bytes32 stakeRoot,
         address chargeRecipient,
         Proof calldata proof
     ) external {
-        // TODO: verify proof
-
-        _postStakeRoot(calculationTimestamp, stakeRoot, chargeRecipient, false);
-    }
-
-    /// @inheritdoc IStakeRootCompendium
-    function blacklistStakeRoot(uint32 submissionIndex) external onlyOwner {
-        // TODO: this should not be onlyOwner
         require(
-            !stakeRootSubmissions[submissionIndex].blacklisted,
-            "StakeRootCompendium.blacklistStakeRoot: stakeRoot already blacklisted"
+            calculationBlockNumber % proofInterval == 0,
+            "StakeRootCompendium._postStakeRoot: blockNumber must be a multiple of proofInterval"
         );
         require(
-            block.timestamp < stakeRootSubmissions[submissionIndex].blacklistableBefore,
-            "StakeRootCompendium.blacklistStakeRoot: stakeRoot cannot be blacklisted"
+            latestBlockNumberProven < calculationBlockNumber,
```

Comment:
`calculationBlockNumber = latestBlockNumberProven + proofInterval`

And this can only be if this is not the first submission

Reply:
what do you mean?

from the first require statement, we have `calculationBlockNumber % proofInterval == 0`

and from `latestBlockNumberProven < calculationBlockNumber`

we have basically that

latestBlockNumberProven = latestBlockNumberProven + n * proofInterval

---

do you mean proofs should be calculated strictly sequentially?